File: tasks/2025/round3_pp/indexing-pp.py
#!/usr/bin/env python3
# @check-accepted: *

def main():
    a = input().strip()
    b = input().strip()

    if a + 'a' == b:
        print(-1)
        return

    a += '0'
    b += '0'

    i = 0
    while True:
        if a[i] != b[i]: break
        i += 1

    if i + 1 < len(b) - 1:
        print(b[:i + 1])
        return

    if ord(b[i]) - ord(a[i]) > 1:
        print(b[:i] + chr(ord(b[i]) - 1))
        return

    i += 1
    while a[i] == 'z':
        i += 1

    print(a[:i] + 'z')

main()

File: tasks/2025/round3_pp/indexing-pp.cpp
// @check-accepted: *
#include <iostream>
#include <string>

int main() {
    std::string a, b;
    std::cin >> a >> b;

    if (a + 'a' == b) {
        std::cout << -1 << '\n';
        return 0;
    }

    std::string c;
    size_t i = 0;
    for (;; ++i) {
        if (a[i] != b[i]) break;
        c += a[i];
    }

    if (i + 1 < b.size()) {
        c += b[i];
        std::cout << c << '\n';
        return 0;
    }

    c += b[i] - 1;
    while (c.back() == a[c.size() - 1]) {
        c += 'z';
    }

    std::cout << c << '\n';
}
